### DISCUSSION TITLE
Task to be refreshed with newly added branch strings - String Based Project

### DISCUSSION THREAD

**Discussion by jimmysafe** [crowdin/crowdin-cli#798] [Q&A]
**Is your feature request related to a problem? Please describe.**
Crowdin Task is associated to a specific branch id. When we add new strings to that branch, the task does not get updated with newly added strings.

**Describe the solution you'd like**
Could we add to the APIV2 an endpoint that gives the possibility to refresh a task strings?

**Additional context**

Notes: 

- We are using string based project
- We are using the APIV3 available endpoints listed here https://developer.crowdin.com/api/v2/string-based/

**Here is an example of our use case:**

**all the following actions are performed through  APIV3 endpoints**

We have **created a branch** called `feat`.

Then we use [Add String](https://developer.crowdin.com/api/v2/string-based/#operation/api.projects.strings.post) Endpoint to add a string to `feat` branch.

Then we use the [Add Task](https://developer.crowdin.com/api/v2/string-based/#operation/api.projects.tasks.post) Endpoint to add a a task and associate `feat` branch to it (the type of task = 1 proofread).

Everything from here looks fine, we have a `feat` branch with a bunch of strings, and we have successfully created a Task which includes those branch strings.

Now if we try to add a new string via [Add String](https://developer.crowdin.com/api/v2/string-based/#operation/api.projects.strings.post) endpoint and we associate the `feat` branch id, **the task which includes all strings from the `feat` branch does not get updated automatically**.

A workaround we have found is to just delete and recreate the task but that is not a very optimal process.

Is there a way to refresh a task when new strings get added to a branch associated to it?

**Comment by andrii-bodnar** [comment]
Hi @jimmysafe, thanks for asking!

Actually, this is the expected behavior. When we create a task, we sort of "lock" its scope and don't automatically update it. It's recommended to create a new task for the new content. I would also recommend using the "Skip strings already included in other tasks" (`skipAssignedStrings`) to include only the new strings in the new task.

You can also [Edit](https://developer.crowdin.com/api/v2/string-based/#operation/api.projects.tasks.patch) the existing task and pass the new list of `stringIds` for translation.
